cirq/_compat.py

```diff
@@ -56,15 +58,18 @@ def deprecated(*, deadline: str, fix: str) -> Callable[[Callable], Callable]:
     def decorator(func: Callable) -> Callable:
         used = False
 
+        @functools.wraps(func)
         def decorated_func(*args, **kwargs) -> Any:
             nonlocal used
             if not used:
                 used = True
+                qualname = (func.__qualname__
+                            if func_name is None else func_name)
                 logging.warning(
                     'DEPRECATION\n'
                     'The function %s was used but is deprecated.\n'
                     'It will be removed in cirq %s.\n'
-                    '%s\n', func.__qualname__, deadline, fix)
+                    '%s\n', qualname, deadline, fix)
 
             return func(*args, **kwargs)
```

Comment:
I added some lines below which set the doc property and such. Is that the purpose of `wraps`? Please resolve the silent merge conflict.

Reply:
Yes.  `wraps` sets `__module__`, `__name__`, `__qualname__`, `__annotations__` and `__doc__` (https://docs.python.org/3/library/functools.html#functools.wraps).

I'll change the lines below to just modify the `__doc__` copied by `wraps`.